### PR TITLE
fix(visualstates): fix rolling back properties when the next state doesn't modify a property, but modifies one of its subproperties

### PR DESCRIPTION
**Pull request by ramezgerges**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/DataBinding/BindingPath.BindingItem.cs [hunk 1]
```diff
@@ -540,6 +538,17 @@ public void NewValue()
...
+
+			/// <remarks>
+			/// This is defined so that 2 BindingItems are equal if they refer to the same property on the same object
+			/// even if they're a part of 2 different "chains".
+			/// </remarks>
+			public bool Equals(BindingItem? that)
+			{
+				return that != null
+					&& !DependencyObjectStore.AreDifferent(this.DataContext, that.DataContext)
+					&& this.PropertyName.Replace(")", "").Replace("(", "").Split(':', '.')[^1] == that.PropertyName.Replace(")", "").Replace("(", "").Split(':', '.')[^1];
```
Comment (ramezgerges) [inline src/Uno.UI/DataBinding/BindingPath.BindingItem.cs:550]: What's blocking this PR is this line. We need to find a way to compare two `BindingItem.PropertyNam`s and figure out if they point to the same thing. E.g. `(Microsoft.UI.Xaml.Controls:Border.Background)` and `Background` are the same thing. There are also expressions with indexers and some other weird stuff.

**Comment by ramezgerges** [comment]
cc. @kazo0, this should fix the Toolkit visual states regression.

**Comment by kazo0** [comment]
fyi @ramezgerges, this original fix seemed to have fixed https://github.com/unoplatform/uno/issues/14262

So rolling it back might unfix it, maybe?

@eriklimakc maybe check from this branch if the issue happens again

**Comment by ramezgerges** [comment]
@kazo0 It shouldn't. The title is a bit misleading :). This PR doesn't "roll back" the previous fix, but it fixes a certain scenario where some properties set by animations should not be rolled back (they get rolled back after the other PR when they shouldn't). So it does indeed move some cases to the old behaviour, but it's very unlikely that that was what fixed #14262.

**Comment by eriklimakc** [comment]
It works fine with the changes from this PR :)

**Review by MartinZikmund — APPROVED** [review]

**Pull request merged**